### PR TITLE
Add parents of parse field to ParseNode.dependentFields

### LINKED ISSUE #3832: Make parse for nested field come after the transform that produces it
- [  ] Fix the issue
- [  ] update the concat flatten example

```
{  "$schema": "https://vega.github.io/schema/vega-lite/v2.json",
  "data": {
    "values": [
      { "id": "001",
        "ra": 243.351,
        "dec": "+54.6",
        "lc": [{"time": 1, "mag": 18.5}, {"time": 2, "mag": 19}]
      },
      { "id": "002",
        "ra": 210.35,
        "dec": "+14.6",
        "lc": [{"time": 1, "mag": 19.5}, {"time": 2, "mag": 20}]
      },
      { "id": "003",
        "ra": 143.35,
        "dec": "+33.6",
        "lc": [{"time": 1, "mag": 19}, {"time": 2, "mag": 18}]
      }
    ]
  },
      "transform": [{"flatten": ["lc"]}],
      "mark": "line",
      "encoding": {
        "x": {
          "field": "lc.time",
          "type": "quantitative",
          "scale": {"zero": false}
        },
        "y": {"field": "lc.mag", "type": "quantitative"}
      }
}
```

### PR DESCRIPTION
**Pull request by invokesus**
Fixes #3832 .

### CODE REVIEW

File: src/compile/data/formatparse.ts [hunk 1]
```diff
@@ -222,7 +222,11 @@ export class ParseNode extends DataFlowNode {
   }
 
   public dependentFields(): StringSet {
-    return toSet(keys(this._parse));
+    const fields = keys(this._parse);
+    const splitFields = fields.map(key => key.split('.'));
```
Comment (domoritz) [inline src/compile/data/formatparse.ts]: Please use the access path function

File: test/compile/data/formatparse.test.ts [hunk 2]
```diff
@@ -307,4 +307,15 @@ describe('compile/data/formatparse', () => {
       assert.deepEqual(p.producedFields(), {n: true, b: true, 'nested.field': true});
     });
   });
+
+  describe('dependentFields', function() {
+    it('should give the correct dependent fields', function() {
+      const p = new ParseNode(null, {
+        d: 'number',
+        'a.b.c': 'boolean'
```
Comment (domoritz) [inline test/compile/data/formatparse.test.ts]: can you test `a['b.c']['d']`?

**Comment by kanitw** [comment]
@invokesus Can you clean this up so we can merge?

**Comment by domoritz** [comment]
Ping @invokesus

File: test/compile/data/formatparse.test.ts [hunk 3]
```diff
@@ -309,13 +309,21 @@ describe('compile/data/formatparse', () => {
...
+      // 'a.b.c' and 'a[b][c]' are both necessary since 'a.b.c' will not be produced from more nested strings
+      assert.deepEqual(p.dependentFields(), {d: true, a: true, 'a[b]': true, 'a[b][c]': true, 'a.b.c': true});
+    });
 
-      assert.deepEqual(p.dependentFields(), {d: true, a: true, 'a.b': true, 'a.b.c': true});
+    it('should give the correct dependent fields for array accessors', function() {
+      const p = new ParseNode(null, {
+        "a['b.c']['d']": 'number'
+      });
+      assert.deepEqual(p.dependentFields(), {a: true, "a[b.c]": true, "a[b.c][d]": true, "a['b.c']['d']": true});
```
Comment (domoritz) [inline test/compile/data/formatparse.test.ts]: `a[b.c]` -> `a['b.c']`
`a[b.c][d]` -> `a['b.c']['d']`

**Comment by invokesus** [comment]
Closing, since we are taking a different approach to this.

**Comment by kanitw** [comment]
Please always delete your branch when you close a PR :)

**Pull request closed without merging**